internal/controller/accountiam_controller.go

```diff
@@ -312,19 +308,37 @@ func (r *AccountIAMReconciler) createOperandRequest(ctx context.Context, instanc
 			},
 		}
 
-		operandReq.SetNamespace(instance.Namespace)
-		if err := controllerutil.SetControllerReference(instance, operandReq, r.Scheme); err != nil {
+		if err := controllerutil.SetControllerReference(instance, operandRequest, r.Scheme); err != nil {
 			return err
 		}
 
-		if err := r.Create(ctx, operandReq); err != nil {
+		if err := r.Create(ctx, operandRequest); err != nil {
 			if !k8serrors.IsAlreadyExists(err) {
 				return err
 			}
 		}
+
+		klog.Infof("Successfully created OperandRequest %s in namespace %s", name, instance.Namespace)
+	} else {
+		// OperandRequest already exists; check and set owner reference if needed
+		needsUpdate := false
+
+		// Check if the owner reference is missing or incorrect
+		if !metav1.IsControlledBy(operandRequest, instance) {
+			klog.V(2).Infof("Setting controller reference for OperandRequest %s", operandRequest.Name)
+			if err := controllerutil.SetControllerReference(instance, operandRequest, r.Scheme); err != nil {
+				return err
+			}
+			needsUpdate = true
+		}
+
+		if needsUpdate {
```

Comment:
I just noticed that we did not have a `createOrupdate` function for OperandRequest? Just curious that is it on purpose?

Reply:
The reason, I think, is that the internal OperandRequest is created by user management only, so there's no need for updating